### DISCUSSION TITLE
pages build and deployment fails with submodules

### DISCUSSION THREAD

**Discussion by justinwyllie** [community/community#111009] [Actions] [answered]
### Select Topic Area

Question

### Body

This job which I didn't know existed started failing recently. It fails at the "build" stage and thus no deployment happens - no files even static ones I upload to the Github pages repository are made available at the site url, though they are in the repo. I don't know what it means by "build" since a) I am just uploading static files and b) as far as I know that is all Github pages supports anyway. In short, there is nothing to "build".  The problem seems to be related to the fact that I commit source code to this repository (the one which is my special Pages repository) and last week I created a Git submodule.  I now get: "clone of 'https://github.com/justinwyllie/SUBMODULE' into submodule path '/home/runner/work/justinwyllie.github.io/justinwyllie.github.io/src/SUBMODULE' fail - 'SUBMODULE, is a .... submodule. It seems that the "build" process (which I still think is not in fact a build process) chokes on a submodule. Is this a known issue? If so, perhaps I should have a separate project for all my code and literally just use github pages for the final JS bundle and html page?

Thanks

**Comment by DeandrePetersen** [comment, marked as answer]
Given your description, it seems that the failure occurs when trying to clone the submodule into the expected directory (/home/runner/work/justinwyllie.github.io/justinwyllie.github.io/src/SUBMODULE). This failure is likely because the submodule path is incorrect or inaccessible.
I would recommend to double-check that the submodule configuration in your main repository is correct. Ensure that the submodule URL is accessible and that the submodule path is correctly specified. If the submodule is causing too much complexity or if it's not essential for your GitHub Pages deployment, consider separating your projects. Have a dedicated repository for your GitHub Pages site, and manage your source code and submodules separately.
Simplify your deployment process by only including the necessary files for your GitHub Pages site in the repository. You can use tools like GitHub Actions to automate the deployment of these files without needing to deal with submodules.

**Reply by justinwyllie** [reply]
The submodules seem to be set up correctly. But, I don't in fact need to commit all the source code to the Pages repository so I broke out the source code into a separate repository and only used the pages one for a few built static files as you suggest. This keeps thing simple.
